Foundation/NSFileManager.swift

```diff
@@ -712,9 +712,15 @@ extension FileManager {
 }
 
 extension FileManager {
-    open var homeDirectoryForCurrentUser: URL { NSUnimplemented() }
+    open var homeDirectoryForCurrentUser: URL {
+        return homeDirectory(forUser: CFCopyUserName().takeRetainedValue()._swiftObject)!
```

Comment:
I wonder if the var declaration could be changed to `open var homeDirectoryForCurrentUser: URL?`. It would make sense for it to be `nil` if it couldn't be obtained by `homeDirectory(forUser:)`